### PR TITLE
Use common code for account retrieval & only require account selection when there is more than one.

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

File: src/commands.rs [hunk 1]
```diff
@@ -17,3 +22,28 @@ pub(crate) mod upgrade;
...
+where
+    DbT::Error: std::error::Error + Sync + Send + 'static,
+{
+    let account_id = match account_uuid {
+        Some(uuid) => Ok(AccountUuid::from_uuid(uuid)),
+        None => {
+            let account_ids = db_data.get_account_ids()?;
+            account_ids.first().map_or(
+                Err(anyhow!(
+                    "More than one account is available; please specify the account UUID."
```
Comment (str4d) [inline src/commands.rs:39]: This seems like it's doing the opposite: if `.first()` returns `None` then there are _no_ accounts, not _several_.

**Review by str4d — CHANGES_REQUESTED** [review]

File: src/commands.rs (hunk 1 above)
Comment (nuttycom) [reply src/commands.rs:39]: :facepalm: I had written this to check `len` and call `first`, and then I was like "oh but I need to error anyway" and forgot to reintroduce the length check.

File: src/commands.rs [hunk 2]
```diff
@@ -17,3 +22,31 @@ pub(crate) mod upgrade;
...
+            let account_ids = db_data.get_account_ids()?;
+            if account_ids.len() > 1 {
+                Err(anyhow!(
+                    "More than one account is available; please specify the account UUID."
+                ))
+            } else {
+                account_ids
+                    .first()
+                    .map_or(Err(anyhow!("Wallet contains no accounts.")), |v| Ok(*v))
+            }
```
Comment (str4d) [inline src/commands.rs:45]: I think slice matching would be cleaner:
```suggestion
            match &account_ids[..] {
                [] => Err(anyhow!("Wallet contains no accounts.")),
                [account_id] => Ok(*account_id),
                _ => Err(anyhow!(
                    "More than one account is available; please specify the account UUID."
                )),
            }
```

**Review by str4d — APPROVED** [review]
utACK 3e529310af2e534eb26a5a2eed3d8aa943636657

**Pull request merged**